src/Illuminate/Validation/Validator.php

```diff
@@ -584,8 +584,12 @@ protected function shouldStopValidating($attribute)
      * @param  array   $parameters
      * @return void
      */
-    protected function addFailure($attribute, $rule, $parameters)
+    public function addFailure($attribute, $rule, $parameters = [])
     {
+        if (! $this->messages) {
+            $this->passes();
```

Comment:
We still need a test for this line.

Reply:
Please help me. Give me a hint on what is expected. My test is asserting that this line is there and it does what it does. What else can I do?

If it was not a `new MessageBag` in the `function passes()`, but instead it was `app()->make(MessageBag...` I could mock it and validate that the passes() function was run. But in this situation I don't see what can be done. Please advise me what to do. Thank you.

Reply:
Sure. The condition needs to be tested, so we need a test where `$messages` is falsey and one where it's true. From what I can tell, you already have a test for when it's true, correct?

Reply:
Actually `messages` neither true nor false it is a way to check if the property exists. The `messages` property is an object of `MessageBag` class. Usually I do checks like this using  `is_null($this->messages)`, but this time, I followed a style of the module.
The function `passes` - is the place where `messages` created and validation performed. It is also not great, because it comes with assumption that if messages are created than validation was performed. So you cannot just create messages without performing validation. And again, I go with the flow, performing validation as expected to create a `MessageBag` object for a `messages` property.

Reply:
@tillkruss , You know what, I can make 2 test cases, one with `messages` created prior to calling addFailure  and one where no messages. I don't see a point doing it, I could not assert much with it, but maybe that is what is expected?

Reply:
@tillkruss , Even better, I can call addFailure  2 times to asert that previous error messages were not lost. Would it count for what is expected?